pkg/auth/providers/azure/azure_provider.go

```diff
@@ -520,21 +520,20 @@ func samePrincipal(me v3.Principal, other v3.Principal) bool {
 }
 
 // UpdateGroupCacheSize attempts to update the size of the group cache defined at the package level.
-func UpdateGroupCacheSize(size string) {
+func UpdateGroupCacheSize(size string) error {
 	if size == "" {
-		return
+		return fmt.Errorf("missing minimum requirements to update group cache size")
 	}
 
 	i, err := strconv.Atoi(size)
 	if err != nil {
-		logrus.Errorf("error parsing azure-group-cache-size, skipping update %v", err)
-		return
+		return fmt.Errorf("error parsing azure-group-cache-size, skipping update %v", err)
 	}
 	if i < 0 {
-		logrus.Error("azure-group-cache-size must be >= 0, skipping update")
-		return
+		return fmt.Errorf("azure-group-cache-size must be >= 0, skipping update")
 	}
```

Comment:
There are no tests for this?

Reply:
If you want to open a separate PR, go for it, equally, it might be easier to keep change and test in the same PR?

Reply:
I'd prefer to keep the two PR separated :)